src/components/transactions/Repay/RepayModalContent.tsx

```diff
@@ -203,11 +203,13 @@ export const RepayModalContent = ({
     );
     const borrowBalanceMarketReferenceCurrency = BigNumber.max(remainingBorrowBalance, 0);
 
-    newHF = calculateHealthFactorFromBalancesBigUnits({
+    const calculatedHealthFactor = calculateHealthFactorFromBalancesBigUnits({
       collateralBalanceMarketReferenceCurrency,
       borrowBalanceMarketReferenceCurrency,
       currentLiquidationThreshold: user?.currentLiquidationThreshold || '0',
-    }).toString(10);
+    });
+
+    newHF = calculatedHealthFactor.isLessThan(0) ? '0' : calculatedHealthFactor.toString(10);
```

Comment:
`calculatedHealthFactor` can equal -1 if the full balance is repaid, which shouldn't be reset, so could you change the condition:

`calculatedHealthFactor.isLessThan(0) && !calculatedHealthFactor.eq(-1)`

Reply:
done!